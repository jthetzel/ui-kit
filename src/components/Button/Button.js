import React from 'react';
import styled from 'react-emotion';

const Button = styled.button`
  background-color: salmon;
  border: 1px solid indianred;
  color: snow;
`;

/**
 * @component
 */
export default props => <Button {...props} />;
