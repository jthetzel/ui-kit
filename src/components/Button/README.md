Button

```js
<Button>Push Me</Button>
```