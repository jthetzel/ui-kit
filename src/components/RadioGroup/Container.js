import styled from 'react-emotion';

export default styled.div`
  display: flex;
  font-family: 'Noto Sans', sans-serif;
  font-size: 12px;
  font-weight: bold;
  margin: auto;
  text-align: center;
`;
