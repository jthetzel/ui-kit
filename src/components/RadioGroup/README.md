RadioGroup

```js
initialState = { value: 1 };

const onChange = (value) => {
  setState({ value });
};

<RadioGroup
  onChange={onChange}
  value={state.value}
>
    <Button value={1}>Option 1</Button>
    <Button value={2}>Option 2</Button>
</RadioGroup>
```