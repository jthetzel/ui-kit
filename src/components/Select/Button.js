import styled from 'react-emotion';

export default styled.div`
  position: relative;
  border-top: 1px solid black;
  border-bottom: 1px solid black;
  padding: 20px 0;
  user-select: none;
  width: 100%;
  
  &:hover {
    cursor: pointer;
  }
`;
