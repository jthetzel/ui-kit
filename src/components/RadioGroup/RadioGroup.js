import PropTypes from 'prop-types';
import React, { Children, createElement } from 'react';

// Style
import Container from './Container';

/**
 * @component
 */

const RadioGroup = ({ value, onChange, children }) => (
  <Container>
    {
      Children.map(children, ({ type, props }) =>
        createElement(type, {
          active: props.value === value,
          onClick: onChange.bind(null, props.value),
          ...props
        }))
    }
  </Container>
);

RadioGroup.propTypes = {
  value: PropTypes.node.isRequired,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export default RadioGroup;
