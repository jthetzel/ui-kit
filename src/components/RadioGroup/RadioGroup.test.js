/* eslint function-paren-newline: 0 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import RadioGroup from './';
import Button from '../Button';

const onChange = value => value;

test('RadioGroup does not crash', () => {
  const tree = shallow(
    <RadioGroup
      onChange={onChange}
      value={1}
    >
      <Button value={1}>Option 1</Button>
      <Button value={2}>Option 2</Button>
    </RadioGroup>);
  expect(toJson(tree)).toMatchSnapshot();
});

test('RadioGroup test value', () => {
  const wrapper = mount(
    <RadioGroup
      onChange={onChange}
      value={1}
    >
      <Button value={1}>Option 1</Button>
      <Button value={2}>Option 2</Button>
    </RadioGroup>);
  expect(wrapper.props().value).toEqual(1);
  wrapper.setProps({ value: 2 });
  expect(wrapper.props().value).toEqual(2);
});
