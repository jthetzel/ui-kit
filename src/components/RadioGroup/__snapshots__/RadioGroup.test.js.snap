// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup does not crash 1`] = `
<Styled(div)>
  <Button
    active={true}
    key=".0"
    onClick={[Function]}
    value={1}
  >
    Option 1
  </Button>
  <Button
    active={false}
    key=".1"
    onClick={[Function]}
    value={2}
  >
    Option 2
  </Button>
</Styled(div)>
`;
